ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/DLModel.java

```diff
@@ -76,10 +79,10 @@ public MLOutput predict(MLInput mlInput) {
         try {
             return AccessController.doPrivileged((PrivilegedExceptionAction<ModelTensorOutput>) () -> {
                 Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
-                if (predictors == null) {
+                if (!isModelReady()) {
```

Comment:
Why changing this condition? It seems they have different logic.

Reply:
Otherwise, It could be a complex logic like ((FunctionName == Tokenizer && tokenizer == null) || (FunctionName != Tokenizer && predictor == null)). Here we want the logic with same as isModelReady except (modelHelper != null && modelId != null), which already been checked before this try. So reuse isModelReady here could be convenient and if we add more model type in the future, we don't need to add logic here.

Reply:
Ok, DLModel has a default implementation for this isModelReady method and it's logic is not changed, in TextEmbeddingSparseEncodingModel, it overrides this method to a different logic. LGTM.